### DISCUSSION TITLE
Oauth / Socialite / Authentication : Please make the instance host for IDP configurable

### DISCUSSION THREAD

**Discussion by nuxwin** [coollabsio/coolify#3713] [Feature Requests]
Good morning,

Coolify is providing authentication through the Socialite Laravel package (oauth) which is really great. However, right now, we cannot setup the instance host for the IdP. For instance, we are using our own GitLab instance and we want use it as IdP. It is not currently possible due to the way you grab the oauth settings. The Laravel Socialite package make that possible easily. For now, we have done something really ugly to test, by setting the host manually...

```php
root@d2b45a9868bb:/var/www/html/bootstrap/helpers# cat socialite.php 
<?php

use App\Models\OauthSetting;
use Laravel\Socialite\Facades\Socialite;

function get_socialite_provider(string $provider)
{
    $oauth_setting = OauthSetting::firstWhere('provider', $provider);

    if ($provider == 'azure') {
        $azure_config = new \SocialiteProviders\Manager\Config(
            $oauth_setting->client_id,
            $oauth_setting->client_secret,
            $oauth_setting->redirect_uri,
            ['tenant' => $oauth_setting->tenant],
        );

        return Socialite::driver('azure')->setConfig($azure_config);
    }

    $config = [
        'client_id' => $oauth_setting->client_id,
        'client_secret' => $oauth_setting->client_secret,
        'redirect' => $oauth_setting->redirect_uri,
    ];

    $provider_class_map = [
        'bitbucket' => \Laravel\Socialite\Two\BitbucketProvider::class,
        'github' => \Laravel\Socialite\Two\GithubProvider::class,
        'gitlab' => \Laravel\Socialite\Two\GitlabProvider::class,
        'google' => \Laravel\Socialite\Two\GoogleProvider::class,
    ];

    $providerInstance = Socialite::buildProvider(
        $provider_class_map[$provider],
        $config
    );

    // We want to use our own GitLab Instance as IdP. (Nuxwin)
    if ($provider == 'gitlab') {
      $providerInstance->setHost('https://private.gitlab-instance.tld');
    }

    return $providerInstance;
}
```

So basically put, a new field (Instance Host) should be added in the UI (default empty), and if provided, should be set on the provider when using it.

Thank you :)

**Comment by rihards-simanovics** [comment]
This is true, GitLab can be hosted locally so having an optional field for GitLab instance i.e. base URL would be useful.
